pkg/token/token.go

```diff
@@ -188,6 +191,21 @@ func (g generator) GetWithRole(clusterID string, roleARN string) (Token, error)
 	if err != nil {
 		return Token{}, fmt.Errorf("could not create session: %v", err)
 	}
+	if g.cache {
+		// figure out what profile we're using
+		var profile string;
+		if v := os.Getenv("AWS_PROFILE"); len(v) > 0 {
+			profile = v
+		} else {
+			profile = session.DefaultSharedConfigProfile
+		}
+		// create a cacheing Provider wrapper around the Credentials
+		if cacheProvider, err := NewFileCacheProvider(clusterID, profile, roleARN, sess.Config.Credentials); err == nil {
```

Comment:
If caching is enabled, wrap the existing Credentials (and its contained Provider) in a new caching provider.